kafka/consumer/group.py

```diff
@@ -567,18 +564,34 @@ def topics(self):
         future = cluster.request_update()
         self._client.poll(future=future)
         cluster.need_all_topic_metadata = stash
+
+    def topics(self):
+        """Get all topics the user is authorized to view.
+
+        Returns:
+            set: topics
+        """
+        cluster = self._client.cluster
+        self._fetch_all_topic_metadata()
         return cluster.topics()
 
     def partitions_for_topic(self, topic):
         """Get metadata about the partitions for a given topic.
+        This method will issue a remote call to the server if it does not already
+        have any metadata about the given topic.
```

Comment:
Let's make it clear in the docstring that there will be times when this method could return values that do not match the state of the cluster due to an outdated cache entry... 

Also, while I have never played with Kafka ACL's, I'm guessing since `_fetch_all_topic_metadata()` only returns info about topics the user is authorized to view that trying to fetch info about a topic that exists but the user can't view will also trigger a remote call / returning `None`? Normally I'd expect a call about a specific topic to return some kind of error from the cluster about the user not authorized to view but here the cluster never sees a topic-scoped call so that's probably not the case.

maybe reword to:

`This method first checks the local metadata cache for information about the topic. If the topic is not found (either because the topic does not exist, the user is not authorized to view the topic, or the metadata cache is not populated), then it will issue a metadata update call to the cluster.`

Reply:
I also have not played with Kafka's ACL's, but that would be my assumption as well.